### PR TITLE
Refactor `initialConsentState`

### PR DESCRIPTION
**Pull request by arelra**
## What does this change?

- Remove internal resolvers from `getInitialConsentState`
- Rename module `initialConsentState` to `initial-consent-state`
- Add unit test

## Does this change need to be reproduced in dotcom-rendering ?

- [x] No
- [ ] Yes (please indicate your plans for DCR Implementation)

### Tested

- [x] Locally
- [x] On CODE (optional)

### CODE REVIEW

**Review by mxdvl — APPROVED** [review]

**Review by zekehuntergreen — APPROVED** [review]

**Review by zekehuntergreen — COMMENTED** [review]
Good call kebab-casing the module name

File: static/src/javascripts/projects/commercial/initial-consent-state.spec.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+		},
+		addtlConsent: 'xyz',
+		gdprApplies: true,
+		tcString: 'YAAA',
+	},
+});
+
+describe('getInitialConsentState', () => {
+	test('tcfv2 with event-status not equal to `cmpuishown` resolves immediately', async () => {
+		const tcfv2ConsentState = builTcfv2ConsentState('tcloaded');
```
[the remark this answers is not in the document]
Comment (arelra) [reply static/src/javascripts/projects/commercial/initial-consent-state.spec.ts:25]: Good shout 👍

File: static/src/javascripts/projects/commercial/initial-consent-state.ts [hunk 2]
```diff
@@ -10,27 +10,20 @@ import type { ConsentState } from '@guardian/consent-management-platform/dist/ty
...
-		if (state.tcfv2) {
-			// For tcfv2 only, the first onConsentChange is fired before the user has
-			// interacted with the consent banner. We want to ignore this first consent state.
-			if (state.tcfv2.eventStatus !== 'cmpuishown')
-				resolveInitialState(state);
-			else return;
-		} else if (state.ccpa || state.aus) {
-			resolveInitialState(state);
-		}
+export const getInitialConsentState = (): Promise<ConsentState> =>
```
[the remark this answers is not in the document]
Comment (arelra) [reply static/src/javascripts/projects/commercial/initial-consent-state.ts:13]: Will do!

But first in order to have some concrete comments I need to firm up that test case we had and work out the potential impact.

I think it's such a subtle edge case, basically when the user is changing their consent state in the same session, between different invocations to `getInitialConsentState`.

I'll do that as a separate task.

**Review by zekehuntergreen — APPROVED** [review]
